src/Microsoft.Health.Fhir.Core/Features/SecretStore/InMemorySecretStore.cs

```diff
@@ -0,0 +1,48 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Core.Features.SecretStore
+{
+    public class InMemorySecretStore : ISecretStore
+    {
+        private Dictionary<string, string> _secrets = new Dictionary<string, string>();
+
+        public Task<SecretWrapper> GetSecretAsync(string secretName)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(secretName);
+
+            if (!_secrets.ContainsKey(secretName))
+            {
+                return null;
```

Comment:
You don't want to return null here since what you want is the result being null or the null task.